### PR TITLE
Solving: undefined method `has_key?'

### PR DESCRIPTION
**Pull request by kalinux**
Hi all,

Using foreman 1.12.1 cause me to get this error, when im trying to get some facts from foreman using lib foreman() in manifest file. I had solving it with this pull request. Always open to hear any suggestions.

Regards
Karen

### CODE REVIEW

File: lib/puppet/parser/functions/foreman.rb [hunk 1]
```diff
@@ -36,6 +36,7 @@
 #
 # Happy Foreman API-ing!
 
+require "pry"
```
Comment (ekohl) [inline lib/puppet/parser/functions/foreman.rb]: I don't think pry is available on every installation.

**Comment by ekohl** [comment]
I'd expect that a timeout would throw an exception rather than returning some invalid data. I wonder if just using pry solves it. It may have some weird side effects that would trigger it.

File: lib/puppet/parser/functions/foreman.rb (hunk 1 above)
Comment (Karen09) [inline lib/puppet/parser/functions/foreman.rb]: if its not available on every installation , how we gonna solve it? maybe adding docs in https://theforeman.org/2012/01/getting-foreman-search-results-into.html ? but other side , as i know "pry" is only debugging so it cant effect for others who`s using this module. is it not?

Comment (mmoll) [inline lib/puppet/parser/functions/foreman.rb]: What is the purpose of this line anyway? IMHO it's simply not possible to *require* a debugger gem for normal production installs (AFAICT that would even needed to get installed in the JRuby Puppetserver if running on that).

The fix for the problem is the increase of the timeout, no?

Comment (ekohl) [inline lib/puppet/parser/functions/foreman.rb]: Looking at the original problem (https://github.com/theforeman/puppet-foreman/issues/482) I find it hard to see how increasing the timeout solves it. That makes me wonder if pry has some sort of side effect.

Comment (mmoll) [inline lib/puppet/parser/functions/foreman.rb]: I see... maybe capturing the failing HTTP request to the foreman API leads to some other hint?

Comment (Karen09) [inline lib/puppet/parser/functions/foreman.rb]: i had tried to put 25 seconds for timeout, somehow it`s does`nt solve problem. after installing pry from first run its work perfectly. Section about  timeout  is less important.

Comment (ekohl) [inline lib/puppet/parser/functions/foreman.rb]: @domcleal any idea why this would happen?

**Comment by ekohl** [comment]
What's the status here?

**Comment by ekohl** [comment]
Closing for inactivity

**Pull request closed without merging**